File: contrib/oracle.pl
#!/opt/lampp/bin/perl
use DBI;

print "Content-Type: text/html\n\n";

my $dsn="dbi:Oracle:host=localhost;sid=xe";
my $dbh=DBI->connect("$dsn","hr","geheim") or
        die "Kann Datenbank nicht erreichen!";

print "<html>";
print "<head>";
print "<title>Perl und Oracle</title>";
print "</head>";
print "<body>";
print "<h1>Perl und Oracle</h1>";
print "<table border=\"1\">";
print "<tr>";
print "<th>Interpret</th>";
print "<th>Titel</th>";
print "<th>Jahr</th>";
print "</tr>";

my $query="SELECT * FROM cds";

my $prep_sql=$dbh->prepare($query) or  die print "Can't prepare";
$prep_sql->execute() or  die print "Can't  execute";

while (my @row = $prep_sql->fetchrow_array())
{
    print "<tr>";
    print "<td>".$row[2]."</td>";
    print "<td>".$row[1]."</td>";
    print "<td>".$row[3]."</td>";
    print "</tr>";
}

$prep_sql->finish();
$dbh->disconnect();

print "</table>";
print "</body>";
print "</html>";

File: contrib/oracle.php
<html>
<head>
<title>PHP und Oracle</title>
</head>
<body>
<h1>PHP und Oracle</h1>
<table border="1">
<tr>
    <th>Interpret</th>
    <th>Titel</th>
    <th>Jahr</th>
</tr>
<?php
    $db="(DESCRIPTION=(ADDRESS_LIST=(ADDRESS=(PROTOCOL=TCP)
       (HOST=localhost) (PORT=1521)))
       (CONNECT_DATA=(SERVICE_NAME=xe)))";

    // Oracle 10g
    //$db="//localhost/xe";

    $c=ocilogon("hr", "geheim", $db);

    $s = ociparse($c, "SELECT * FROM cds");

    if(ociexecute($s))
    {
        while (ocifetch($s)) 
        {
            echo "<tr>";
            echo "<td>".ociresult($s, "INTERPRET")."</td>";
            echo "<td>".ociresult($s, "TITEL")."</td>";
            echo "<td>".ociresult($s, "JAHR")."</td>";
            echo "</tr>";
        }
    }
    else
    {
	$e = oci_error($s); 
        echo htmlentities($e['message']);
    }
?>
</table>
</body>
</html>

